src/RoadRunnerBridge/HttpFoundationWorker.php

```diff
@@ -188,7 +179,141 @@ private function timeFloat(): float
+     * @return \Closure
+     */
+    private function createFileStreamGenerator(BinaryFileResponse $response): \Closure
+    {
+        return static function () use ($response) {
+            $ref = new \ReflectionClass($response);
+            $maxlen = $ref->getProperty("maxlen")->getValue($response);
+            $offset = $ref->getProperty("offset")->getValue($response);
+            $chunkSize = $ref->getProperty("chunkSize")->getValue($response);
+            $deleteFileAfterSend = $ref->getProperty("deleteFileAfterSend")->getValue($response);
+
+            try {
+                if (!$response->isSuccessful()) {
+                    return;
+                }
+
+                $file = fopen($response->getFile()->getPathname(), "r");
+
+                ignore_user_abort(true);
+
+                if ($maxlen === 0) {
+                    return;
+                }
+
+                if ($offset !== 0) {
+                    fseek($file, $offset);
+                }
+
+                $length = $maxlen;
+                while ($length && !feof($file)) {
+                    $read = $length > $chunkSize || 0 > $length ? $chunkSize : $length;
+
+                    if (false === $data = fread($file, $read)) {
+                        break;
+                    }
+
+                    while ('' !== $data) {
+                        try {
+                            yield $data;
+                        } catch (StreamStoppedException) {
+                            break 2;
+                        }
+
+                        if (0 < $length) {
+                            $length -= $read;
+                        }
+                        $data = substr($data, $read);
+                    }
+                }
+
+                fclose($file);
+            } finally {
+                if ($deleteFileAfterSend && is_file($response->getFile()->getPathname())) {
+                    unlink($response->getFile()->getPathname());
+                }
+            }
+        };
+    }
+
+    /**
+     * @param SymfonyResponse $response
+     * @return \Closure
+     */
+    private function createDefaultContentGetter(SymfonyResponse $response): \Closure
+    {
+        return static function () use ($response) {
+            ob_start();
+            $response->sendContent();
+            return ob_get_clean();
+        };
+    }
+
+    /**
+     * StreamedResponse callback can now use `yield` to be really streamed
+     * @param StreamedResponse $response
+     * @return \Closure
+     */
+    private function createStreamedResponseGenerator(StreamedResponse $response): \Closure
+    {
+        return function () use ($response): \Generator {
+            $kernelCallback = $response->getCallback();
+
+            $kernelCallbackRef = new \ReflectionFunction($kernelCallback);
+            $closureVars = $kernelCallbackRef->getClosureUsedVariables();
+
+            $ref = new \ReflectionFunction($closureVars["callback"]);
```

Comment:
Your idea isn't a bad solution, but the likelihood of them refactoring this part of code or changing listener priorities to mess up your idea of the implementation will be probably the same. Don't forget tho, that these events can be `stopPropagation()` and then we will have to fall back to some other method of getting the callback - also low likelihood but the possibility is still there. This is really up to you.

PS: why would `RR_MODE` matter? if it's anything else then `http`, the listener will not be triggered anyway AFAIK

Reply:
I think it will depends on what we do here: https://github.com/Baldinof/roadrunner-bundle/pull/130#discussion_r1418028234

If we add a custom `StreamedResponse` we can just extend the regular `Response` and Symfony will not call `setCallback()` to wrap it.

Reply:
`RR_MODE` or any other var that indicate we are not running with RR and should consume the generator and echo it